### ISSUE TITLE
Create an equivalent external/integration API endpoint for /api/status

### ISSUE THREAD

**Issue by Tournesol-me** [owncast/owncast#3981]
### Share your bug report, feature request, or comment.

Would be useful to have a get handle on /api/integrations/streamtitle for third party clients

**Comment by gabek** [comment]
The `/api/status` endpoint is unauthenticated, you should be able to pull it from there.

**Comment by Tournesol-me** [comment]
Oh you are absolutely right !

But on the [api doc, /api/status](https://owncast.online/api/latest/#tag/Internal/operation/GetStatus) says viewerCount instead of streamTitle

{
  "serverTime": "string",
  "lastConnectTime": "string",
  "lastDisconnectTime": "string",
  "versionNumber": "string",
  "viewerCount": 0,
  "online": true
}

just a heads up !

**Comment by gabek** [comment]
That's probably because a stream title isn't set.

[Here is an example](https://party.radiofreefedi.net/api/status)


```json
{"serverTime":"2024-10-24T19:26:30.201092317Z","lastConnectTime":"2024-10-23T21:52:16Z","lastDisconnectTime":null,"versionNumber":"20240404-nightly","streamTitle":"RFF COMFY CHANNEL SIMULCAST","online":true}

```

**Comment by Tournesol-me** [comment]
yes, could see it but you only can discover it by chance if it isn't in the doc, no ?

**Comment by gabek** [comment]
The API is in the docs, but that's true, it doesn't have that property documented.

Alternatively, you can use the config endpoint, and that does have the property documented: https://owncast.online/api/latest/#tag/Internal/operation/GetWebConfig

<blockquote><div><strong><a href="https://owncast.online/api/latest/#tag/Internal/operation/GetWebConfig">Owncast APIs</a></strong></div></blockquote>

**Comment by Tournesol-me** [comment]
thank you for the help, I'll use those alternatives for the time being !

**Comment by apshada** [comment]
@gabek I would like to work on this issue, can you please assign it to me

**Comment by gabek** [comment]
@apshada Sure! You should be able to add a new endpoint and call the same handler as the existing API, no need to rebuild it. In the future if there's a breaking change then the internal and external payloads can be split up.

**Comment by varungujarathi9** [comment]
Hey @gabek 
Can I work on this if it hasn't started yet?

**Comment by gabek** [comment]
@apshada Are you still looking at this?

**Comment by germainelee** [comment]
Bump

**Comment by RizqiSugiarto** [comment]
Hi, I noticed this issue is still open. Would it be okay if I worked on it?  I also have a couple of questions I'd like to clarify.

**Comment by gabek** [comment]
@RizqiSugiarto Sure! The logic for this API response already exists, so it's just a new endpoint pointing to it.

https://owncast.notion.site/API-Web-Routing-Development-43f9c6a2f35143fabb6b2036ee6f09ef

Let me know what questions you have!

<blockquote><img src="https://www.notion.so/images/meta/default.png" width="48" align="right"><div>Notion</div><div><strong><a href="https://www.notion.so">Notion – The all-in-one workspace for your notes, tasks, wikis, and databases.</a></strong></div><div>A new tool that blends your everyday work apps into one. It's the all-in-one workspace for you and your team</div></blockquote>

**Comment by RizqiSugiarto** [comment]
@gabek As I understand it, I just need to generate a new endpoint, and in that new handler, I just need to use the same core as the get status API, is that correct?

**Comment by gabek** [comment]
Yup, that's right! You can just fire the existing handler. https://github.com/owncast/owncast/blob/develop/webserver/handlers/status.go#L16

<blockquote><img src="https://repository-images.githubusercontent.com/266445736/4f82fb80-a5b1-11ea-8f1c-69d316f861b5" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/owncast/owncast/blob/develop/webserver/handlers/status.go">owncast/webserver/handlers/status.go at develop · owncast/owncast</a></strong></div><div>Take control over your live stream video by running it yourself.  Streaming + chat out of the box. - owncast/owncast</div></blockquote>

**Comment by gabek** [comment]
@RizqiSugiarto Are you still looking at this?
